### PR TITLE
[evaluation] Add evaluate api missing input support

### PR DESCRIPTION
**Pull request by diondrapeck**
# Description

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by MilesHolland — APPROVED** [review]

**Review by nagkumar91 — APPROVED** [review]

File: sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/_evaluate/_evaluate.py [hunk 1]
```diff
@@ -287,7 +287,9 @@ def _validate_columns_for_evaluators(
                 # Ignore the missing fields if "conversation" presents in the input data
                 missing_inputs = []
             else:
-                missing_inputs = [col for col in evaluator_params if col not in new_df.columns]
+                optional_params = evaluator._OPTIONAL_PARAMS if hasattr(evaluator, "_OPTIONAL_PARAMS") else []
+                excluded_params = set(new_df.columns).union(optional_params)
```
Comment (ninghu) [inline sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/_evaluate/_evaluate.py:291]: Would this call fail if new_df.columns already contains "query"?

**Review by ninghu — APPROVED** [review]

File: sdk/evaluation/azure-ai-evaluation/tests/e2etests/data/evaluate_test_data_no_query.jsonl [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{"context":"AML API only","response":"To create a run using the Azure Machine Learning API, you first need to create an Experiment. Once you have an experiment, you can create a Run object that is associated with that experiment. Here is some Python code that demonstrates this process:\n\n```\nfrom azureml.core import Experiment, Run\nfrom azureml.core.workspace import Workspace\n\n# Define workspace and experiment\nws = Workspace.from_config()\nexp = Experiment(workspace=ws, name='my_experiment')\n\n# Create a new run\nrun = exp.start_logging()\n```\n\nIn this code, the `from_config()` method reads the configuration file that you created when you set up your Azure Machine Learning workspace. The `Experiment` constructor creates an Experiment object that is associated with your workspace, and the `start_logging()` method creates a new Run object that is associated with the Experiment. Now you can use the `run` object to log metrics, upload files, and track other information related to your machine learning experiment.", "ground_truth":"Paris is the capital of France."}
```
Comment (w-javed) [inline sdk/evaluation/azure-ai-evaluation/tests/e2etests/data/evaluate_test_data_no_query.jsonl:1]: Shouldn't it be messages 

For example
https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/evaluation/azure-ai-evaluation/tests/unittests/data/generated_qa_chat_conv.jsonl

GroundednessPro is using 
https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/evaluation/azure-ai-evaluation/tests/e2etests/data/evaluate_test_data_conversation.jsonl

Comment (diondrapeck) [reply sdk/evaluation/azure-ai-evaluation/tests/e2etests/data/evaluate_test_data_no_query.jsonl:1]: I was using this for composite tests (i.e. GroundednessEvaluator and F1ScoreEvaluator) and F1ScoreEvaluator doesn't support conversation, so an explicit column mapping format needs to be used for input.

Comment (diondrapeck) [reply sdk/evaluation/azure-ai-evaluation/tests/e2etests/data/evaluate_test_data_no_query.jsonl:1]: I've added a test to test on the same dataset as groundednesspro as well.

**Pull request closed without merging**